File: kennedy-langlotz/KLQ/src/com/klq/typechecker/error/CyclicDependency.java
package com.klq.typechecker.error;

import com.klq.ast.impl.QuestionNode;

/**
 * Created by Juriaan on 1-3-2015.
 */
public class CyclicDependency extends AError{
    public CyclicDependency(QuestionNode node) {
        super(7, true, String.format("There is a cyclic dependency at the following question: %s", node.getQuestionID()), node.getLocation());
    }
}

File: FelipezConde/QuestionnaireLanguage/TypeChecking/Notifications/Errors/CyclicDependency.cs
using AST.Nodes.Expressions;

namespace TypeChecking.Notifications.Errors
{
    public class CyclicDependency : Error
    {
            Id id;

            public CyclicDependency(Id id)
            {
                this.id = id;
            }

            public override string Message()
            {
                return string.Format("Cyclic dependency for identifier \"{0}\" at {1}",
                                      new object[] {id.Name,
                                                    id.GetPosition()
                                                   });
            }
    }
}
